packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,199 @@
+					   @Inject(UIRouter) public routingService: UIRouter) {
+		this.pathCtrl = new FormControl();
+	}
+	
+	/**
+	 * Filters the options according to what has been entered by the user in the input field
+	 */
+	public filterOptions(value: string): RouteEntry[] {
+		const filterValue: string = value.toLowerCase();
+		return this.optionsToDisplay.filter((state: RouteEntry) => state.label.toLowerCase().indexOf(filterValue) === 0);
+	}
+
+	
+	public ngOnInit(): void {
+
+		this.inputValue="";
+		this.filteredOptions = this.pathCtrl.valueChanges.pipe(
+			startWith(""),
+			map((option: string) => (option ? this.filterOptions(option) : this.optionsToDisplay.slice()))
+		);
+		
+		this.hide = true;
+		
+		this.logger.debug(componentName + ": component initialized");
+
+		if (this.optionsToDisplay !== undefined){
+			if (this.optionsList === undefined){
+				this.optionsToDisplay =	this.initRoutePathFromRouter();
+			} else{
+				this.optionsToDisplay = this.initRoutePathFromConfig();
+			}
+		}
+		this.sortPathList();
+		this.inputValue="";
+	}
+
+	/**
+	 * redirect the user to the chosen path
+	 */
+	public redirect(value: string): void {
+		this.inputValue="";
+		this.starkRoutingService.navigateTo(value);
+	}
+
+	/**
+	 * Show/hide the input field
+	 */
+	public show(): void {
+		this.hide = !this.hide;
+	}
+	/**
+	 * @ignore
+	 */
+	public trackPath(_index: number, path: RouteEntry): string {
+		if (path.targetState !== undefined){
+			return path.targetState;
+		} else 
+		{
+			return "";
+		}
+	}
+
+	/**
+	 * Retrieve the list of routes from a StarkMenuConfig object which the user have passed to the component
+	 */
+	//tslint:disable:cognitive-complexity
+	public initRoutePathFromConfig(): RouteEntry[]{
+		const optionsToDisplay: RouteEntry[] = [];
+		if (this.optionsList !== undefined && this.optionsList.menuSections !== undefined) {
+			this.optionsList.menuSections.forEach((section: StarkMenuSection) => {
+				section.menuGroups.forEach((group: StarkMenuGroup) => {
+					if (group.isVisible && group.isEnabled && group.targetState) {
+						optionsToDisplay.push({
+							label: group.label,
+							targetState: group.targetState,
+							targetStateParams: group.targetStateParams
+						});
+					}
+					if (group.entries) {
+						group.entries.forEach((entry: StarkMenuGroup) => {
+							if (entry.isEnabled && entry.isVisible) {
+								optionsToDisplay.push({
+									label: entry.label,
+									targetState: entry.targetState,
+									targetStateParams: entry.targetStateParams
+								});
+							}
+						});
+					}
+				})
+			});
+		}
+		return optionsToDisplay;
+	}
+
+	/**
+	 * Retrieve the list of routes from UIRouter if the user haven't passed a list to the component
+	 */
+	public initRoutePathFromRouter(): RouteEntry[]{
+		const optionsToDisplay: RouteEntry[] = [];
+		if (this.optionsList === undefined){
+			for (const state of this.routingService.stateService.get()) {
```

Comment:
For this logic, we should also think about the lazy loaded modules to avoid loading the modules while trying to fetch the routes!